### PR TITLE
[17.09] Migrate documentation for setting up a production instance to Galaxy.

### PR DESCRIPTION
**Pull request by jmchilton**
This is a bugfix for the soon to be introduced bug of missing documentation for 17.09 and before. 18.01 will require a significant rewrite of these pages for the uWSGI and configuration changes to Galaxy. Since we are now hosting older branches of these docs including the [release_17.09](https://docs.galaxyproject.org/en/release_17.09/), backporting this configuration documentation will allow us to provide permanent links to this 17.09 variant of all this documentation (with the final copy of information about Paste and galaxy.ini).

This content currently migrates 8 pages of important admin documentation.

The following 4 pages have been migrated to the main admin section of the Galaxy docs:

- the production configuration page
- the cluster configuration page
- the job configuration page
- the scaling information page

The following 4 pages have been migrated to the "special topics" admin section of the Galaxy docs:

- the ftp configuration page
- the top-level nginx page (for external auth content still linking out to the hub)
- the top-level apache page (for external auth content still linking out to the hub)
- the job metrics configuration page

This PR also includes small tweaks to sphinx documentation generation.

- Add a quick build mode that skips source generation / Python docs (export GALAXY_DOCS_SKIP_SOURCE=1 to build this way). This speeds up review of core Sphinx docs.
- Use AutoStructify for better Markdown integration.

Additionally this PR includes updates to this documentation above and beyond what is on the corresponding Hub pages to reflect more of the state of art as of 17.09 including:

- Update the job configuration page with information about XML macros.
- Mention and link to newer style resubmission condition expressions (not just an enumerated string anymore).
- Updated list of dynamic job runner configuration parameters.
- Mention the ``raw`` attribute for ``env`` directives in the job configuration page.
- Mention the ``total_walltime`` limit added in #3217.
- Drop mentions of older URL-based job conf configuration stuff.
- Mention newer Local and Paramiko based shell runners in CLI section.
- Mention newer slurm job styles in CLI section.
- Improved Pulsar links.
- Fix dataset security link in Apache display sites config.
- Update job metrics page to link to a new Github issue instead of old Trello issues.

### CODE REVIEW

File: doc/source/admin/jobs.md [hunk 1]
```diff
@@ -0,0 +1,403 @@
+# Galaxy Job Configuration
+
+By default, jobs in Galaxy are run locally on the server on which the Galaxy application was started.  Many options are available for running Galaxy jobs on other systems, including clusters and other remote resources.
+
+This document is a reference for the job configuration file.  [Detailed documentation](cluster) is provided for configuring Galaxy to work with a variety of Distributed Resource Managers (DRMs) such as TORQUE, Grid Engine, LSF, and HTCondor.  Additionally, a wide range of infrastructure decisions and configuration changes should be made when running Galaxy as a production service, as one is likely doing if using a cluster.  It is highly recommended that the [production server documentation](production.html) and [cluster configuration documentation](cluster.html) be read before making changes to the job configuration.
+
+**The most up-to-date details of advanced job configuration features can be found in the [sample job_conf.xml](https://github.com/galaxyproject/galaxy/blob/dev/config/job_conf.xml.sample_advanced) found in the Galaxy distribution.**
+
+### Galaxy Configuration
```
Comment (hexylena) [inline doc/source/admin/jobs.md]: level 1 -> level 3?

**Comment by hexylena** [comment]
:+1:

**Review by hexylena — APPROVED** [review]

**Comment by bgruening** [comment]
That's super cool, I was worried about this how we can document the new changes and keep the old ones. And this is clearly the way to go! I also like to keep the docs close to the source, I think we discussed this before but never did this move. Big thanks @jmchilton! This IS awesome!

**Review by bgruening — APPROVED** [review]

**Review by martenson — APPROVED** [review]
Great step in the direction we want to go - provide readthedocs-like experience for version-bound documentation. Next piece of the puzzle would probably be an actual DOM element that provides the version selector.

I think this is very important for the Oslo admin training and labeling missing docs as a bug seems appropriate.

Thank you John.

**Pull request merged**